src/codegen/MHTranslator.cpp

```diff
@@ -1036,24 +1036,92 @@ void MHTranslator::transQuery(code::FunctionDecl* fun, std::shared_ptr<ir::Query
           (qr->getVar()->getTyp()->getTyp() == ir::IRConstant::BOOL ?
             BOOL_TYPE : mapIRTypeToCodeType(qr->getVar()->getTyp())) })),
           initArgs);
-  code::VarDecl::createVarDecl(
-      coreNs, answervarname,
-      code::Type(HISTOGRAM_CLASS_NAME, std::vector<code::Type>( {
+  // Everything same up until here
+  if (qr->hasForLoop()) {
+    transForloopQueryHelper(fun, qr, n, initvalue);
```

Comment:
@giulio-zhou I still think the transForloopQueryHelper() can share lots of code with the following else branch.

I think you can merge these two branches to make things nicer.